pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -39,7 +42,32 @@ def _ops_equal(op1, op2):
     )
 
 
-def _are_inverses(op1, op2):
+def _check_abstractness(op1: Operator, op2: Operator) -> bool:
+    """Checks if either of the operators has abstract wires, parameters, or hyperparameters."""
+
+    def contains_abstract(items):
+        """Checks if any item in the iterable is abstract."""
+        for item in items:
+            if is_abstract(item):
+                return True
+            if isinstance(item, Wires) and any(is_abstract(w) for w in item):
+                return True
+        return False
+
+    def op_has_abstract(op):
+        """Checks if the operator has abstract wires, parameters, or hyperparameters."""
+        return any(
+            [
+                contains_abstract(op.wires),
+                hasattr(op, "parameters") and contains_abstract(op.parameters),
+                hasattr(op, "hyperparameters") and contains_abstract(op.hyperparameters.values()),
```

Comment:
Also, we should *never* have an abstract hyperparameter. If we do, a lot of things are going to break, not just this transform.